server/dash/pages/nc_summary_comparison.py

```diff
@@ -104,11 +358,325 @@ def generate_dynamic_filter(selected_nc):
         neigbhorhood council.
     """
     # If no neighborhood council is selected, use all data.
-    print("got new selected_nc ", selected_nc)
     if not selected_nc:
         df = api_data_df
     else:
         df = api_data_df[api_data_df["councilName"] == selected_nc]
 
     req_types = sorted(list(set(df["typeName"])))
     return req_types, ' '
+
+# TODO: Add request type filter if necessary? 
+@callback(
+    Output("nc_avg_comp_line_chart", "figure"),
+    [Input("selected_nc", "value")]
+)
+def update_line_chart(selected_nc):
+    """Generates a line chart visualizations for LA 311 requests data
+     based on the two selected neighborhood conucils.
+    This function takes the selected neighborhood council (nc) value
+     from the "selected_nc" dropdown and output a line chart showing 
+    the number of requests throughout the day and the average number 
+    of requests throughout the day (total number of requests / all 99 neighborhood councils).
+    Args:
+        selected_nc: A string argument automatically detected by Dash 
+        callback function when "selected_nc" element is selected in the layout.
+    Returns: 
+        nc_avg_comp_line_chart: line chart showing the number of requests
+         throughout the day for the selected neighborhood council and average
+    """
+    # If dropdown value is empty, use all data available.
+    if not selected_nc:
+        df = report_json
+        selected_nc = 'Total'
+    else:
+        df = report_json[report_json.council_name == selected_nc]
+
+    # Calculating the average number of requests throughout the day.
+    neighborhood_sum_df = df.groupby(["created_date"]).agg("sum").reset_index()  # noqa
+    total_sum_df = report_json.groupby(["created_date"]).agg("sum").reset_index()
+    total_sum_df["nc_avg"] = total_sum_df["counts"] / 99
+    merged_df = neighborhood_sum_df.merge(total_sum_df["nc_avg"].to_frame(),
+     left_index=True, right_index=True)  # noqa
+
+    nc_avg_comp_line_chart = px.line(
+        merged_df,
+        x="created_date",
+        y=["counts", "nc_avg"],
+        color_discrete_sequence=DISCRETE_COLORS,
+        labels=LABELS,
+        title="Number of " + selected_nc +
+        " Requests compared with the average Neighborhood Council"
+    )
+
+    nc_avg_comp_line_chart.update_xaxes(
+        tickformat="%a\n%m/%d",
+    )
+
+    nc_avg_comp_line_chart.update_traces(
+        mode="markers+lines"
+    )  # add markers to lines.
+
+    return nc_avg_comp_line_chart
+
+
+@callback(
+    Output("req_type_pie_chart", "figure"),
+    Input("selected_nc", "value"),
```

Comment:
Are the Inputs supposed to be together in a list?

Reply:
Yeah because we want ensure multi-selection is possible.